girdermedviewer/app/girder/utils.py

```diff
@@ -14,24 +14,27 @@ class CacheMode(Enum):
     Permanent = "Permanent"
 
 
-class FileDownloader:
-    def __init__(self, girder_client, temp_dir=None, cache_mode=CacheMode.No):
+class FileFetcher:
+    def __init__(self, girder_client, assetstore_dir=None, temp_dir=None, cache_mode=CacheMode.No):
         """
         :example:
         ```
         girder_client = GirderClient(apiUrl="http://localhost:8080/api/v1")
         girder_client.authenticate(apiKey="my_key")
-        downloader = FileDownloader(girder_client, cache_mode=CacheMode.Session)
-        with downloader.download_file(file) as file_path:
+        fetcher = FileFetcher(girder_client, cache_mode=CacheMode.Session)
+        with fetcher.fetch_file(file) as file_path:
             ...
         ```
         """
+        self.assetstore_dir_path = assetstore_dir
         self.temp_dir_path = temp_dir
+        if temp_dir == assetstore_dir:

```

Comment:
does it work if one path is relative and the other one is absolute ? you should probably use an os.path function to compare paths